blaze/compute/numpy.py

```diff
@@ -198,8 +198,23 @@ def compute_up(t, x, **kwargs):
 
 
 @dispatch(Distinct, np.ndarray)
-def compute_up(t, x, **kwargs):
-    return np.unique(x)
+def compute_up(t, arr, **kwargs):
+    if t.on:
+        if getattr(arr.dtype, 'names', None) is not None:
+            return pd.DataFrame.from_records(arr).drop_duplicates(
+                subset=t.on if t.on else None,
+            ).reset_index(drop=True).value.astype(arr.dtype)
```

Comment:
i think you might not be covering this in tests since `value` isn't an attribute of `DataFrame`s